discovery-provider/src/queries/get_health.py

```diff
@@ -529,6 +548,47 @@ def get_user_bank_health_info(
     }
 
 
+def get_reactions_health_info(
+    redis: Redis,
+    max_indexing_drift: Optional[int] = None,
+    max_reaction_drift: Optional[int] = None,
+):
+    now = datetime.now()
+    last_index_time = redis.get(LAST_REACTIONS_INDEX_TIME_KEY)
+    last_index_time = int(last_index_time) if last_index_time else None
+    last_reaction_time = redis.get(LAST_SEEN_NEW_REACTION_TIME_KEY)
+    last_reaction_time = int(last_reaction_time) if last_reaction_time else None
+
+    last_index_time = (
+        datetime.fromtimestamp(last_index_time) if last_index_time else None
+    )
+    last_reaction_time = (
+        datetime.fromtimestamp(last_reaction_time) if last_reaction_time else None
+    )
+
+    indexing_delta = (
+        (now - last_index_time).total_seconds() if last_index_time else None
+    )
+    reaction_delta = (
+        (now - last_reaction_time).total_seconds() if last_reaction_time else None
+    )
+
+    is_unhealthy_indexing = bool(
+        indexing_delta and max_indexing_drift and indexing_delta > max_indexing_drift
+    )
+    is_unhealthy_reaction = bool(
+        reaction_delta and max_reaction_drift and reaction_delta > max_reaction_drift
```

Comment:
Not sure how useful this one will be for determining overall health, since it's possible nobody reacts. I see it's an optional param for now though but still curious if it needs to be part of the overall is_healthy calculation?

Reply:
I was thinking you can set it super conservatively - like if nobody reacts in 2 days or something. Without something like this, the indexing check alone won't catch if there's a logic error somewhere later in that function that processes the reactions. Wdyt

Reply:
My biggest concern would be a spiralling reselection and degraded user experience because nobody reacted for a while, but as long as we don't include the argument in libs and keep it optional, I'm ok with it

Reply:
Good thought - would definitely only include this for our monitoring system